### PR TITLE
chore: upgrade GolangCI Lint to v1.52.0

### PR DESCRIPTION
**Pull request by xcoulon**
also, fix issues

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

**Review by MatousJobanek — APPROVED** [review]

**Review by mfrancisc — APPROVED** [review]
Thanks for cleaning it up! 

I left only a small comment.

File: controllers/useraccount/useraccount_controller_test.go [hunk 1]
```diff
@@ -1862,7 +1862,7 @@ func gockFindUserTimes(name string, times int, calls *int) { //nolint: unparam
 		BodyString(fmt.Sprintf(`{"name":"%s","id":"abc1234"}`, name))
 }
 
-func gockFindUserNoBody(name string, code int, calls *int) { //nolint: unparam
+func gockFindUserNoBody(_ string, code int, calls *int) { //nolint: unparam
```
[the remark this answers is not in the document]
Comment (xcoulon) [reply controllers/useraccount/useraccount_controller_test.go:1865]: yes, indeed! Done in ea0e405

[the remark this answers is not in the document]
Comment (mfrancisc) [reply controllers/useraccount/useraccount_controller_test.go]: Thanks for addressing my comment!

**Pull request merged**